### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by panpeterone** [NVIDIA-Digital-Bio/Proteina-Complexa#29]
Thanks for this protein design tool.

### **Question / Potential Bug Report:**

Missing 1D sequence conditioning for target hotspots due to `target_hotspot_mask` vs `hotspot_mask` mismatch?

### **Description & Observation**

I am running a binder generation pipeline using the `target_hotspots` parameter (e.g., in `targets_dict.yaml` with `hotspot_residues: ["A235", "A237"]`).

While my job ran, I noticed the following warning in `generate.log` (logged only once at the beginning):

```text
WARNING  No hotspot_mask in batch, returning zeros for HotspotMaskSeqFeat     seq_cond_feats.py:348
```

After checking the generation trajectory, it visually appears that the binder is drawn to the target region (possibly guided by 2D spatial pair features like `CrossSequenceHotspotMaskPairFeat`). However, it feels like the model struggles to form a chemically sound interface (high clash, very low RF3 ipTM score, e.g. ~0.1), as if the 1D sequence conditioning track might be completely oblivious to the intended hotspot constraints.

### **My Investigation & Question**

I took a brief look through the source code and found what looks like a variable naming disconnect:

1. In `script/src/proteinfoundation/datasets/gen_dataset.py`, `TargetFeatures.__call__` parses the hotspots and explicitly assigns them to the key `target_hotspot_mask`:

   ```python
   if self.target_hotspots is not None and self.target_hotspots_masks is not None:
       result["target_hotspot_mask"] = self.target_hotspots_masks[sample_idx].clone()
   ```
2. However, in `script/src/proteinfoundation/nn/feature_factory/seq_cond_feats.py`, `HotspotMaskSeqFeat` statically expects the key `hotspot_mask`:

   ```python
   def forward(self, batch):
       if "hotspot_mask" in batch: 
           ...
       else:
           # Hits the else branch every time and returns torch.zeros
           if not self._has_logged:
               logger.warning("No hotspot_mask in batch...")
   ```

Because the sequence feature factory evaluates `HotspotMaskSeqFeat` and expects `batch["hotspot_mask"]`, it seems to be receiving an all-zero tensor instead of the actual hotspot mask.

**My questions for the developers:**

1. Is this `No hotspot_mask in batch` warning an expected behavior under certain configurations, or is it a bug?
2. Does this variable mismatch mean the global 1D sequence track is indeed generating blindly without the hotspot constraint, even though the 2D pair track receives `target_hotspot_mask` correctly?
3. If this is a bug, could it be resolved by simply mirroring the mask in `gen_dataset.py` (e.g., adding `result["hotspot_mask"] = result["target_hotspot_mask"]`), or is there a specific dataset transform I should be invoking in the YAML config to properly bridge this gap?

Thank you for your time and guidance!

**Comment by nvdreidenbach** [comment]
The hotspot warning is due to deprecated code. the hotspot features are used on the concat feature factory via TargetConcatSeqFeat that uses HotspotMaskSeqFeat that uses hotspot_mask but from 


hotspot_feats = self.hotspot_feat(
                {
                    "hotspot_mask": batch["target_hotspot_mask"],
                }
            ) 


. We did not use any explicit hotspot features elsewhere so would you do not need to change anything.
